Remove index page links from doc status

Elchi3 opened the pull request:
To get rid of index pages, we want to rip them out of doc status.
This is a minimal changes PR, because I don't want to touch anything here and doc status will be retired later anyway, too.

wbamberg commented:
Thanks @Elchi3 !

I've tested the `DocStatus` change on "/en-US/docs/MDN/Doc_status/JavaScript" in a local Kuma and it works fine. It might be nicer if we changed https://github.com/mdn/kumascript/blob/master/macros/DocStatus.ejs#L320-L329 so if `link` is undefined, we don't create an `<a>` element. As it is, we create an `<a>` with an empty href, which looks broken.

I couldn't find anywhere `DocStatusSummary is used`...

Elchi3 commented:
> It might be nicer if we changed https://github.com/mdn/kumascript/blob/master/macros/DocStatus.ejs#L320-L329 so if link is undefined, we don't create an <a> element. As it is, we create an <a> with an empty href, which looks broken.

Good point, I've added a commit that does that.

> I couldn't find anywhere DocStatusSummary is used...

Apparently on https://wiki.developer.mozilla.org/de/docs/MDN/Doc_status/API and some other translations of that page.
